pulsar-client-cpp/python/pulsar/schema/definition.py

```diff
@@ -114,34 +121,50 @@ def schema_info(cls, defined_names):
 
         defined_names.add(namespace_name)
 
-        schema = {'name': str(cls.__name__)}
+        schema = {
+            'type': 'record',
+            'name': str(cls.__name__)
+        }
         if cls._avro_namespace is not None:
             schema['namespace'] = cls._avro_namespace
-        schema['type'] = 'record'
         schema['fields'] = []
 
-        for name in sorted(cls._fields.keys()):
+        if cls._sorted_fields:
+            fields = sorted(cls._fields.keys())
+        else:
+            fields = cls._fields.keys()
+        for name in fields:
             field = cls._fields[name]
             field_type = field.schema_info(defined_names) \
                 if field._required else ['null', field.schema_info(defined_names)]
-            schema['fields'].append({
-                'name': name,
-                'type': field_type,
-                'default': field.default()
-            }) if field.required_default() else schema['fields'].append({
-                'name': name,
-                'type': field_type,
-            })
+
+            if field._required_default:
+                schema['fields'].append({
+                    'name': name,
+                    'default': None,
```

Comment:
if `field.default()=_empty_default` the `'default': None`, if `field.default()!`=`_empty_default` the `default': field.default()`